ceno_zkvm/src/tables/mod.rs

```diff
@@ -18,18 +18,23 @@ pub use program::{InsnRecord, ProgramTableCircuit};
 mod ram;
 pub use ram::*;
 
-pub trait TableCircuit<E: ExtensionField> {
+pub trait TableCircuit<E: ExtensionField>
+where
+    Self: Default,
```

Comment:
This Default requirement could move as far out as possible towards the function that actually needs it.